orttraining/orttraining/python/training/ortmodule/_utils.py

```diff
@@ -99,6 +99,18 @@ def get_device_from_module(module):
         pass
     return device
 
+
+def get_device_from_inputs(args, kwargs):
```

Comment:
This change looks good to me, but it raised a question: do we have any tests for a model with inputs on different devices?

(I'm not 100% sure if that's expected to work, or if it raises a useful error message.)

Reply:
We have this check: https://github.com/microsoft/onnxruntime/blob/f97764432430cef27dfd6782338604ed1f0a0a14/orttraining/orttraining/python/training/ortmodule/_training_manager.py#L32

A test was earlier added for this:
https://github.com/microsoft/onnxruntime/blob/f97764432430cef27dfd6782338604ed1f0a0a14/orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py#L1546